packages/backend-core/src/__tests__/utils/request.test.ts

```diff
@@ -0,0 +1,79 @@
+import { checkCrossOriginReferrer } from '../../util/request';
+
+describe('check cross-origin-referrer request utility', () => {
+  test('is not CO with IPv6', () => {
+    const referrerURL = new URL('http://[::1]');
+    const host = new URL('http://[::1]').host;
+    expect(checkCrossOriginReferrer({ referrerURL, host })).toEqual(false);
+  });
+
+  test('is not CO with set https and 443 port', () => {
+    const referrerURL = new URL('https://localhost:443');
+    const host = new URL('https://localhost').host;
+    expect(checkCrossOriginReferrer({ referrerURL, host })).toEqual(false);
+  });
+
+  test('is CO with mixed default security ports', () => {
+    const referrerURL = new URL('https://localhost:80');
+    const host = new URL('http://localhost:443').host;
+    expect(checkCrossOriginReferrer({ referrerURL, host })).toEqual(true);
+  });
+
+  test.skip('is CO when HTTPS to HTTP', () => {
```

Comment:
@colinclerk @SokratisVidros 
I think we cannot safely cover this case (_and vice versa_) as is.

Reply:
I think you're right, so this is best effort.

One thing I forgot is that we also try to detect `crossOriginReferrer` with the check immediately above this:
```js
    if (isDevelopmentOrStaging(API_KEY) && !clientUat) {
      return {
        status: AuthStatus.Interstitial,
        interstitial: await fetchInterstitial(),
      };
    }
```

This works because `clientUat` has `SameSite=Strict` in development, so if the page navigation is Cross**Site** then this cookie won't exist and we'll throw an interstitial.

The `checkCrossOriginReferrer` we're working on tries to capture the requests that are Cross**Origin** but not Cross**Site**.  For example, imagine a developer running their own version of `accounts` on `localhost:4000` then redirecting the user to `localhost:3000`.

It makes sense to me that Referrer-Policy can prevent this detection, but I think our options are:
1. Leave the relatively narrow error cases
2. Throw an interstitial when there's no `Referer`, but this breaks `view-source`

Reply:
I would say we are ok with this. _(at least until the next revision)_